File: app/src/main/java/com/person/v_plaunov/mylistview/CoinParser.java
package com.person.v_plaunov.mylistview;

import org.xmlpull.v1.XmlPullParser;
import java.util.ArrayList;

public class CoinParser {
    private ArrayList<Coin> coins;

    public CoinParser() {
        coins = new ArrayList<>();
    }

    public ArrayList<Coin> getCoins() {
        return coins;
    }

    public boolean parse(XmlPullParser xpp) {
        boolean status = true;
        Coin currentCoin = null;
        boolean inEntry = false;
        String textValue = "";

        try {
            int eventType = xpp.getEventType();
            while (eventType != XmlPullParser.END_DOCUMENT) {

                String tagName = xpp.getName();
                switch (eventType) {
                    case XmlPullParser.START_TAG:
                        if ("mony".equalsIgnoreCase(tagName)) {
                            inEntry = true;
                            currentCoin = new Coin(null,null,null,null,null, null);
                        }
                        break;
                    case XmlPullParser.TEXT:
                        textValue = xpp.getText();
                        break;
                    case XmlPullParser.END_TAG:
                        if(inEntry){
                            if("mony".equalsIgnoreCase(tagName)){
                                coins.add(currentCoin);
                                inEntry = false;
                            } else if("id".equalsIgnoreCase(tagName)){
                                currentCoin.setCoinId(textValue);
                            } else if("name".equalsIgnoreCase(tagName)){
                                currentCoin.setCoinNominal(textValue);
                            }
                        }
                        break;
                    default:
                }
                eventType = xpp.next();
            }
        }
        catch (Exception e){
            status = false;
            e.printStackTrace();
        }
        return  status;
    }
}

File: app/src/main/java/com/person/v_plaunov/mylistview/CoinParser.kt
package com.person.v_plaunov.mylistview

import org.xmlpull.v1.XmlPullParser

class CoinParser {
    val coins: ArrayList<Coin?>

    init {
        coins = ArrayList()
    }

    fun parse(xpp: XmlPullParser): Boolean {
        var status = true
        var currentCoin: Coin? = null
        var inEntry = false
        var textValue: String? = ""
        try {
            var eventType = xpp.eventType
            while (eventType != XmlPullParser.END_DOCUMENT) {
                val tagName = xpp.name
                when (eventType) {
                    XmlPullParser.START_TAG -> if ("mony".equals(tagName, ignoreCase = true)) {
                        inEntry = true
                        currentCoin = Coin(null, null, null, null, null, null)
                    }
                    XmlPullParser.TEXT -> textValue = xpp.text
                    XmlPullParser.END_TAG -> if (inEntry) {
                        if ("mony".equals(tagName, ignoreCase = true)) {
                            coins.add(currentCoin)
                            inEntry = false
                        } else if ("id".equals(tagName, ignoreCase = true)) {
                            currentCoin!!.coinId = textValue
                        } else if ("name".equals(tagName, ignoreCase = true)) {
                            currentCoin!!.coinNominal = textValue
                        }
                    }
                    else -> {}
                }
                eventType = xpp.next()
            }
        } catch (e: Exception) {
            status = false
            e.printStackTrace()
        }
        return status
    }
}